node/impl/full/eth.go

```diff
@@ -904,9 +911,62 @@ func (a *EthModule) EthTraceBlock(ctx context.Context, blkNum string) ([]*ethtyp
 		}
 	}
 
+	var ethTraces []*ethtypes.EthTrace
+	for _, trace := range allTraces {
+		trace := trace
+		ethTraces = append(ethTraces, trace.EthTrace)
+	}
+
+	if err := a.updateContractAddresses(ctx, ts, ethTraces, deployedCode); err != nil {
+		return nil, xerrors.Errorf("failed to update contract addresses: %w", err)
+	}
+
 	return allTraces, nil
 }
 
+func (a *EthModule) updateContractAddresses(ctx context.Context, ts *types.TipSet, allTraces []*ethtypes.EthTrace, deployedCode map[abi.ActorID]*cid.Cid) error {
+	// Update Contract Addresses if contract still exists
+	for _, trace := range allTraces {
+		switch r := trace.Result.(type) {
```

Comment:
This isn't correct, we want to update the last one _only_. The idea was to record a pointer to `r.Code` in `deployedCode`, so we could set it without even having to look at the traces. That is:

```go
deployedCode := make(map[abi.ActorID]*EthBytes)
```

Then, inside the `buildTraces` logic, we'd record `deployedCode[receiverID] = &r.Code`.

Finally, we'd iterate over `deployedCode` and set `*deployedCode[receiverID] = loadByteCode(receiverID)`, as long as the receiver is still live.